File: src/main/java/com/lessmarkup/interfaces/system/MailSender.java
package com.lessmarkup.interfaces.system;

import java.util.OptionalLong;

public interface MailSender {
    <T extends MailTemplateModel> void sendMail(Class<T> type, String smtpServer, String smtpUser, String smtpPassword, boolean smtpSsl, String emailFrom, String emailTo, String viewPath, T parameters);
    <T extends MailTemplateModel> void sendMail(Class<T> type, String emailFrom, String emailTo, String viewPath, T parameters);
    <T extends MailTemplateModel> void sendMail(Class<T> type, OptionalLong userIdFrom, OptionalLong userIdTo, String userEmailTo, String viewPath, T parameters);
    void sendPlainEmail(String emailTo, String subject, String message);
}

File: src/main/scala/com/lessmarkup/interfaces/system/MailSender.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.system

trait MailSender {
  def sendMail[T <: MailTemplateModel](modelType: Class[T], smtpServer: String, smtpUser: String, smtpPassword: String, smtpSsl: Boolean, emailFrom: String, emailTo: String, viewPath: String, parameters: T)

  def sendMail[T <: MailTemplateModel](modelType: Class[T], emailFrom: String, emailTo: String, viewPath: String, parameters: T)

  def sendMail[T <: MailTemplateModel](modelType: Class[T], userIdFrom: Option[Long], userIdTo: Option[Long], userEmailTo: String, viewPath: String, parameters: T)

  def sendPlainEmail(emailTo: String, subject: String, message: String)
}